app/assets/javascripts/katello/providers/provider_redhat.js

```diff
@@ -55,7 +55,7 @@ KT.redhat_provider_page = (function($) {
         } else {
             options['repo'] = "0";
         }
-
+        options['pulp_id'] = checkbox.attr("data-pulp-id");
```

Comment:
I don't really like passing the pulp id here, as we shouldn't be using that to uniquely identify the repository.  We should be using the upstream name IMHO.  NACK to this.